packages/ng-query/src/lib/tests/entity-utils.spec.ts

```diff
@@ -0,0 +1,28 @@
+import { TestBed } from '@angular/core/testing';
+import { QueryClientService } from '../query-client';
+import { arrayPush } from '../entity-utils';
+
+interface Item {
+  id: string;
+}
+
+describe('entity utils', () => {
+  it('should add entity to the query', () => {
+    const { client } = setup();
+    const item1: Item = { id: 'item-1' };
+    const item2: Item = { id: 'item-2' };
+
+    client.setQueryData(['item'], item1);
+    client.setQueryData(['item'], arrayPush('item', item2));
```

Comment:
@NetanelBasal started with writing test of exisiting method `arrayPush` before making any change to signature. 

It does not pass the test. The [`state`](https://github.com/ngneat/query/blob/dfc9e81a3efb49e36e34ab6df69bf5e00b3d339a/packages/ng-query/src/lib/entity-utils.ts#L8) passed by `QueryClient` is not the List, instead its first query-item and hence it breaks saying that `state` is not iterateable.